src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -986,10 +986,10 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     try {
                         X509Certificate[] x509Certificates = X500.asX509CertificateArray(sslCallback.getSslSession().getPeerCertificates());
                         verifyEvidence(new X509PeerCertificateChainEvidence(x509Certificates));
```

Comment:
Maybe it would be better to instead modify the process like this:

- In the SSL handler block, cache the server certs into a serverCerts field, and attempt to acquire the client certs into a clientCerts field (set it to null if the peer is unverified)
- In the authorization handler block, if there is no enclosing identity, check the client certificates (which should be acquired during handling of the SSL callback as today, just stored in a clientCerts field) using verifyEvidence to establish the enclosing identity; if verify fails, return an unauthorized response.  Then proceed with authorization as normal.